File: TAbaretata/CSharpPartTwo-2015/StringsAndTextProcessing/04.SubstringInText/SubstringInText.cs
using System;

/*  Write a program that finds how many times a sub-string is contained in a given text (perform case insensitive search).
    Example:
    The target sub-string is in
    The text is as follows: We are living in an yellow submarine. We don't have anything else. 
    inside the submarine is very tight. So we are drinking all the day. We will move out of it in 5 days.
 */

class SubstringInText
{
    static void Main()
    {
        string text = @"We are living in an yellow submarine. We don't have anything else. 
            Inside the submarine is very tight. So we are drinking all the day. We will move out of it in 5 days.";
        string sub = "in";
        int count = 0;
        for (int i = 0; i < text.Length - 1; i++)
        {
            if (text.Substring(i, sub.Length).ToLower() == sub)
            {
                count++;
                i += sub.Length;
            }
        }
        Console.WriteLine(count);
    }
}

File: TAbaretata/JavaScript-Fundamentals-2015/08.Strings/03.SubstringInText/SubstringInText.js
/*
Problem 3: Write a JavaScript function that finds how many times a substring is contained in a given text 
(perform case insensitive search).
Example: The target substring is "in". The text is as follows:

We are living in an yellow submarine. We don't have anything else. Inside the submarine is very tight. 
So we are drinking all the day. We will move out of it in 5 days.

The result is: 9.
*/

function substringCount(text) {
    var splitedText = text.toLowerCase().split(substrWord),
		wordCount = splitedText.length - 1;
    console.log(wordCount <= 0 ? 'Word not found' : 'Word counts: ' + wordCount);
}

var text = "We are living in an yellow submarine. We don't have anything else. Inside the submarine is very tight. So we are drinking all the day. We will move out of it in 5 days.",
    substrWord = 'in';
substringCount(text);

